chromatic_fitting.py

```diff
@@ -257,46 +418,27 @@ def attach_data(self, r):
         Rainbow object
         """
         self.data = r._create_copy()
-        self.apply_operation_to_constituent_models(LightcurveModel.attach_data, r)
+        self.apply_operation_to_constituent_models('attach_data', r)
+
+    def choose_optimization_method(self, optimization_method='simultaneous'):
+        LightcurveModel.choose_optimization_method(self, optimization_method)
+        if optimization_method == "separate":
+            self.apply_operation_to_constituent_models('change_all_priors_to_Wavelike')
 
     def setup_orbit(self):
```

Comment:
Is this necessary anymore in the `CombinedModel`, since `TransitModel` already makes sure that the orbit gets set up inside of `setup_lightcurves`?

Reply:
Probably not, though it depends whether you want to keep the functionality to setup the orbit separately to the lightcurves (which we can do in the TransitModel